src/lib.rs

```diff
@@ -160,6 +163,22 @@ impl Pty {
           return Err(Error::new(ErrorKind::Other, "setsid"));
         }
 
+        // set the cgroup if specified
+        #[cfg(target_os = "linux")]
+        if let Some(cgroup_path) = &opts.cgroup_path {
+          let pid = libc::getpid();
+          let cgroup_path = format!("{}/cgroup.procs", cgroup_path);
+          let mut cgroup_file = File::create(cgroup_path)?;
+          cgroup_file.write_all(format!("{}", pid).as_bytes())?;
+        }
+        #[cfg(not(target_os = "linux"))]
+        if opts.cgroup_path.is_some() {
```

Comment:
it'd be fun to move this to the very top of the function, so that we don't do a lot of work and fail at the last possible moment, ha.